### ISSUE TITLE
Inconsistency in how AB-3 and AB-4 depict associations vs dependencies

### ISSUE THREAD

**Issue by DrWala** [nus-cs2103-AY2021S2/forum#305]
In the AB-3 and AB-4 Model Component of the DG, both diagrams maintain an association between ModelManager and Person as seen here:

AB3:
![](https://se-education.org/addressbook-level3/images/ModelClassDiagram.png)
AB4:
![](https://se-education.org/addressbook-level4/images/ModelClassDiagram.png)

The logical assumption here would be that ModelManager has many Persons, via the FilteredList.

However, both their storage components depict dependencies (and not associations) between JsonSerializableAddressBookStorage and JsonSerializablePerson / JsonSerializableAddressBook and JsonAdaptedPerson

AB3:
![](https://se-education.org/addressbook-level3/images/StorageClassDiagram.png)
AB4:
![](https://se-education.org/addressbook-level4/images/StorageClassDiagram.png)

The JsonSerializableAddressBook similarly maintains a list of JsonAdaptedPersons.

So the question is: Why does the storage treat this relationship as a dependency but the model treats it as an association?

You could argue that Storage does not directly interact with JsonAdaptedPerson—it does so via the List object, and therefore it's a dependency as it's a typed list. On the other hand, you could argue that we should not view the List as an object and view it just as a container, and thus its a 1:* relationship.

May I know what the module's take on this is?

**Comment by damithc** [comment]
Good point. I think Storage should use associations instead. For JasonAdaptedTag as well.

**Comment by DrWala** [comment]
Alright thank you. Should I close this issue, and is this a bug for AB3 to fix?

**Comment by damithc** [comment]
> Should I close this issue, and is this a bug for AB3 to fix?

Let's leave it open for some time.

**Comment by tlylt** [comment]
One of the testers pointed out an inaccuracy in our team's model diagram:
- `ReadOnlyUserPrefs` should be marked as `<<interface>>`
Though a bit late, would like to just confirm that this is a valid inherited bug from AB3 (and seems like AB4 as well)? 
![image](https://user-images.githubusercontent.com/41845017/115655313-a0384a00-a365-11eb-92d4-6ee19c006639.png)

**Comment by damithc** [comment]
> One of the testers pointed out an inaccuracy in our team's model diagram:
> 
> * `ReadOnlyUserPrefs` should be marked as `<<interface>>`
>   Though a bit late, would like to just confirm that this is a valid inherited bug from AB3 (and seems like AB4 as well)?
>   ![image](https://user-images.githubusercontent.com/41845017/115655313-a0384a00-a365-11eb-92d4-6ee19c006639.png)

Thanks for letting us know @tlylt 
Will convey this to the AB3 team, to be fixed before the next round.
